org.adempiere.base/src/org/compiere/model/GridTable.java

```diff
@@ -424,8 +424,10 @@ private String createSelectSql()
 			m_SQL += " ORDER BY " + m_orderClause;
 		}
 		//
-		log.fine(m_SQL_Count);
-		Env.setContext(m_ctx, m_WindowNo, m_TabNo, GridTab.CTX_SQL, m_SQL);
+		if (log.isLoggable(Level.FINE))
+			log.fine(m_SQL_Count);
+		if (log.isLoggable(Level.INFO))

```

Comment:
Having this in context is very useful to find sometimes why the expected records are not appearing - but as the workaround is now to set the log level on INFO, I think is OK and we can live with that.

Reply:
Yes, very  usefull for debug but we shouldn't have that in production. Performance aside, those sql can include sensitive data that user shouldn't see.